src/BenchmarkDotNet/Jobs/NugetReferenceList.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace BenchmarkDotNet.Jobs
+{
+    /// <summary>
+    /// An ordered list of Nuget references. Does not allow duplicate references with the same PackageName.
+    /// </summary>
+    internal class NugetReferenceList : IReadOnlyCollection<NugetReference>
```

Comment:
hmm, maybe we should give up on the idea of `NugetReference` and use `Dictionary<string, string>` instead?